### ISSUE TITLE
`mad_tpsa_atan2`, `mad_ctpsa_polar`, `mad_ctpsa_carg` incorrect in Quadrants 2, 3

### ISSUE THREAD

**Issue by mattsignorelli** [MethodicalAcceleratorDesign/MAD-NG#415]
For example, going around the unit circle with the scalars as TPSAs with `mad_tpsa_atan2`:
```
atan2(0,1) = 0.0
atan2(1,0) = 1.5707963267948966
atan2(0,-1) = 1.5707963267948966   # Should be pi
atan2(-1,0) = -1.5707963267948966
```

And:
```
atan2(3,2) =  9.8279372324732905  # Quadrant 1
atan2(-3,2) = -9.8279372324732905 # Quadrant 4
atan2(3,-2) = 2.5535900500422257  # Quadrant 2 - incorrect, should be 2.1587989303424644
atan2(-3,-2) = -2.5535900500422257 # Quadrant 3 - incorrect, should be -2.1587989303424644
```

`mad_ctpsa_carg` and `mad_ctpsa_polar` similarly are incorrect in Quadrants 2 and 3

**Comment by ldeniau** [comment]
I confirm that atan2 is completely buggy resulting from a quick and dirty implementation... I will look for the best way to fix it, in particular, how to keep the proper derivatives when x=0. Other cases are just to do proper quadrant management.

**Comment by ldeniau** [comment]
fixed in 413ee9bc
